bokehjs/src/coffee/models/tools/draw/box_draw_tool.ts

```diff
@@ -0,0 +1,96 @@
+import {Dimensions} from "core/enums"
+import * as p from "core/properties"
+import {ColumnDataSource} from "models/sources/column_data_source"
+import {SelectTool} from "../gestures/select_tool"
+import {EditToolView} from "./edit_tool"
+
+
+export interface BkEv {
+  bokeh: {
+    sx: number
+    sy: number
+  }
+  srcEvent: {
+    shiftKey?: boolean
+  }
+  keyCode: number
+  timeStamp: number
+}
+
+
+export class BoxDrawToolView extends EditToolView {
+  model: BoxDrawTool
+  _basepoint: number[]
```

Comment:
Should be a tuple, so `_basepoint: [number, number]`.

Reply:
Value-level syntax is the same, thus `[1, 1]` can be `number[]` as well as `[number, number]`, so explicit type annotations may be necessary later in the code to differentiate between those two cases.

Reply:
Thanks, I puzzled over this one briefly, I really appreciate your help in bringing me up to speed with the syntax.